src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Unix.cs

```diff
@@ -137,14 +138,22 @@ internal SocketError ReplaceHandle()
 
             // Then replace the handle with a new one
             SafeSocketHandle oldHandle = _handle;
-            SocketError errorCode = SocketPal.CreateSocket(_addressFamily, _socketType, _protocolType, out _handle);
+            SocketError errorCode = SocketPal.CreateSocket(_addressFamily, _socketType, _protocolType, out SafeSocketHandle newHandle);
+            Volatile.Write(ref _handle, newHandle);
```

Comment:
@tmds what is the reason you wanted to move the replacement to the end of the method? I don't see any issue with it being done here.

Note that we need to update the handle so we can touch the `Socket` properties for using new handle.

Reply:
In fact, this was the reason why CI failed.

Reply:
I suggested to move it at the end because that ensures all initialization of the handle has happened when we assign it to the field.

> Note that we need to update the handle so we can write Socket properties using the new handle.

Ah yes, I didn't think of that.